sycl/test-e2e/bindless_images/vulkan_interop/vulkan_common.hpp

```diff
@@ -271,10 +302,31 @@ VkResult setupDevice(const sycl::device &dev) {
                                  });
           return (it != std::end(supportedDeviceExtensions));
         });
+    // Skip this device if it does not support all required extensions.
     if (!hasRequiredExtensions) {
       continue;
     }
 
+    // Check if the device supports the optional extensions, if so add them to
+    // the list of enabled device extensions.
+    std::for_each(std::begin(optionalExtensions), std::end(optionalExtensions),
+                  [&](const char *optionalExt) -> void {
```

Comment:
```suggestion
    for (const char *optionalExt : optionalExtensions) {
```

In most cases I find range-based for loops simpler than `std::for_each`